evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -72,9 +72,15 @@ where
     /// The aggregation circuit, which verifies two proofs that can either be
     /// root or aggregation proofs.
     pub aggregation: AggregationCircuitData<F, C, D>,
+    /// The two-to-one aggregation circuit, which verifies two unrelated
+    /// aggregation proofs.
+    pub two_to_one_aggregation: TwoToOneAggCircuitData<F, C, D>,
     /// The block circuit, which verifies an aggregation root proof and an
     /// optional previous block proof.
     pub block: BlockCircuitData<F, C, D>,
+    /// The two-to-one block aggregation circuit, which verifies two unrelated
+    /// block proofs.
+    pub two_to_one_block: TwoToOneAggCircuitData<F, C, D>,
```

Comment:
What exactly is the goal behind? This will only support one additional layer of aggregation, not an arbitrary one, which I would have thought was the desired goal?

Reply:
I think so too. I think this is just a temporary solution. @muursh ?

Reply:
Aye it is just a place holder for now. We will have Einar make some modifications to the current approach.